server/src/internalClusterTest/java/org/elasticsearch/snapshots/CorruptedBlobStoreRepositoryIT.java

```diff
@@ -760,28 +763,81 @@ public void testSnapshotWithMissingShardLevelIndexFile() throws Exception {
             Files.delete(shardGen);
         }
 
-        logger.info("--> creating another snapshot");
+        if (randomBoolean()) {
+            logger.info(
+                "--> restoring the snapshot, the repository should not have lost any shard data despite deleting index-N, "
+                    + "because it uses snap-*.data files and not the index-N to determine what files to restore"
```

Comment:
Nit: can we not use `index-N` for shard generation files? I find it better to reseve it for the root blob. So maybe `index-UUID`? Old snapshots do use numeric but that is not the case in this test.

Reply:
++ eb233fff308139826c337362092e5751a6f3de89